test/services/StreamLabs.spec.js

```diff
@@ -2,16 +2,28 @@ const axios = require('axios');
 const { StreamLabs } = require('../../src/services/StreamLabs');
 
 describe('StreamLabs', () => {
+	describe('#constructor', () => {
+		it('takes an optional logger argument', () => {
+			const subject = new StreamLabs({ token: '' }, { log: jest.fn() });
+
+			expect(subject).toBeInstanceOf(StreamLabs);
+		});
+	});
 	describe('#alert', () => {
 		let axiosSpy;
 
-		beforeEach(() => {
-			jest.restoreAllMocks();
+		const setupDefaultAxiosSpy = () => {
 			axiosSpy = jest.spyOn(axios, 'post');
 			axiosSpy.mockImplementationOnce(() => {});
+		};
+
+		afterEach(() => {
+			jest.restoreAllMocks();
 		});
 
```

Comment:
On the HTTP 401 test, the previous `beforeEach `with cleanup + mock an empty callable on `axios.post` didn't fit the test.
Instead of forcing every test with `beforeEach`, I refactored it to use " explicit setup".

See slides 50,51 https://www.slideshare.net/PHPVigoGrupo/php-vigotalk-unit-testing-sucks-and-its-your-fault